doc/developer-guides/coding_guidelines.rst

```diff
@@ -1787,6 +2323,197 @@ Compliant example::
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       struct struct_showcase
+       {
+           uint32_t *temp_32;
+           uint64_t *temp_64;
+       };
+       
+       uint32_t *showcase_ptr_u32;
+       struct struct_showcase *showcase_ptr_struct;
+       
+       showcase_ptr_u32 = (uint32_t *)showcase_ptr_struct;
+
+
+TY-23: Assignment on function pointers shall be performed with same type
+========================================================================
+
+Compliant example::
+
+    typedef void (*func_ptr_t)(void);
+    
+    func_ptr_t func_ptr_a;
+    func_ptr_t func_ptr_b;
+    
+    func_ptr_a = func_ptr_b;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       typedef void (*func_ptr_a_t)(void);
+       typedef uint32_t (*func_ptr_b_t)(uint32_t param);
+       
+       func_ptr_a_t func_ptr_a;
+       func_ptr_b_t func_ptr_b;
+       
+       func_ptr_a = func_ptr_b;
+
+
+TY-24: Cast shall not be performed on function pointer
+======================================================
+
+Compliant example::
+
+    typedef uint32_t (*func_ptr_t)(uint32_t param);
+    
+    uint32_t func_showcase(uint32_t param)
+    {
+        return param;
+    }
+    
+    func_ptr_t func_ptr_showcase;
+    func_ptr_showcase = func_showcase;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       typedef uint32_t (*func_ptr_t)(uint32_t param);
+       
+       void func_showcase(uint32_t param)
+       {
+           printf("param: %d \n", param);
+       }
+       
+       func_ptr_t func_ptr_showcase;
+       func_ptr_showcase = (func_ptr_t)func_showcase;
+
+
+TY-25: A string literal shall only be used as const object
+==========================================================
+
+The following operations shall be covered:
+
+a) If a string literal is assigned to a variable, this variable shall be
+   declared with const qualifier;
+b) If a string literal is passed as a function parameter, this function
+   parameter shall be declared with const qualifier;
+c) If a string literal is used as the return value of a function, this function
+   return type shall be declared with const qualifier.
+
+Compliant example::
+
+    const char *showcase = "showcase";
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       char *showcase = "showcase";
+
+
+TY-26: The basic numerical types shall not be used other than in typedefs
+=========================================================================
+
+Typedef-name shall be used to replace the usage of basic numerical types. This
+is to guarantee the code portability between different compilers and platforms.
+
+Compliant example::
+
+    typedef unsigned int uint32_t;
+    
+    uint32_t showcase = 32U;
+
+.. rst-class:: non-compliant-code
+
+   Non-compliant example::
+
+       unsigned int showcase = 32U;
+
+
+TY-27: The operands type of the assignment operator shall be the same
```

Comment:
This would read better as ``The operands of an assignment operator shall be the same type``